tests/scripts/run_op_profiling.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from loguru import logger
+import numpy as np
+import pytest
+
+from tt_metal.tools.profiler.process_model_log import run_device_profiler, post_process_ops_log
+
+
+from tt_metal.tools.profiler.common import PROFILER_LOGS_DIR, PROFILER_DEVICE_SIDE_LOG
+
+profiler_log_path = PROFILER_LOGS_DIR / PROFILER_DEVICE_SIDE_LOG
+
+from tt_metal.tools.profiler.process_device_log import import_log_run_stats
+import tt_metal.tools.profiler.device_post_proc_config as device_post_proc_config
+
+matmul_baseline_kernel_duration_gs = [
+    347076.298,
+    656080.448,
+    347153.226,
+    655997.51,
+    175263.62,
+    328744.596,
+    175286.458,
+    328779.454,
+    1966421.516,
+    2192647.532,
+    2176973.452,
+    2250058.658,
+]
+
+matmul_baseline_kernel_duration_wh = [
+    230737,
+    329652,
+    417351,
+    419615,
+    214276,
+    304323,
+    393762,
+    394062,
+    148655,
+    191725,
+    245358,
+    247829,
+    124888,
+    158895,
+    222026,
+    222966,
+    892214,
+    1301073,
+    1593991,
+    1601608,
+    802177,
+    1169392,
+    1476336,
+    1472074,
+    597053,
+    776933,
+    996587,
+    985680,
+    478471,
+    613935,
+    # 928427,
+    # 907950,
+]
+
+
+def run_op_test():
+    logger.info(f"========= RUNNING OP TEST - matmul ")
+
+    op_name = "tt::operations::primary::Matmul"
+    duration_cols = ["DEVICE KERNEL DURATION [ns]"]
+
+    run_device_profiler(
+        "pytest tests/tt_eager/python_api_testing/unit_testing/misc/test_matmul_1d_2d.py", "op_profiler_results"
+    )
+    results = post_process_ops_log("op_profiler_results", duration_cols, False, op_name)
+    kernel_durations_ns = results[duration_cols[0]]
+
+    setup = device_post_proc_config.default_setup()
+    setup.deviceInputLog = profiler_log_path
+    deviceData = import_log_run_stats(setup)
+    freq = deviceData["deviceInfo"]["freq"]
```

Comment:
Oh I see you still needed to get freq and revert back to cycles. This will bring precision loss but it should be ok for this test as it is well outside your 3% margin.